drivers/rtc/rtc_emul.c

```diff
@@ -376,32 +362,30 @@ static int rtc_emul_alarm_get_time(const struct device *dev, uint16_t id, uint16
 		return -EINVAL;
 	}
 
-	k_mutex_lock(&data->lock, K_FOREVER);
-
-	(*timeptr) = data->alarms[id].datetime;
-	(*mask) = data->alarms[id].mask;
-
-	k_mutex_unlock(&data->lock);
+	K_SPINLOCK_GUARDED_SECTION(&data->lock)
```

Comment:
Same - do it the old fashioned way, but please do either make an issue or a PR for the guarded macro.

Reply:
Ignored for now as not being part of this PR - see discussion below.